internal/handlers/common/projection.go

```diff
@@ -221,28 +221,264 @@ func projectDocumentWithoutID(doc *types.Document, projection *types.Document, i
 			)
 
 		case *types.Array, string, types.Binary, types.ObjectID,
-			time.Time, types.NullType, types.Regex, types.Timestamp: // all this types are treated as new fields value
+			time.Time, types.NullType, types.Regex, types.Timestamp: // all these types are treated as new fields value
 			projected.Set(key, value)
 
 		case bool: // field: bool
-			// process top level fields
-			if path.Len() == 1 {
-				if inclusion {
-					if docWithoutID.Has(key) {
-						projected.Set(key, must.NotFail(docWithoutID.Get(key)))
-					}
-
-					continue
+			if inclusion {
+				// inclusion projection copies the field on the path from docWithoutID to projected.
+				if _, err = includeProjection(path, docWithoutID, projected); err != nil {
+					return nil, err
```

Comment:
How do we process this error? As far as I understand, this error is `pathNotExists`, but I don't see any place where we compare with it or return a specific error code based on it. Also, this row is marked as not covered by tests.